src/mono/mono/mini/mini-llvm.c

```diff
@@ -11276,6 +11276,33 @@ MONO_RESTORE_WARNING
 		}
 #endif
 
+#ifdef TARGET_AMD64
+		case OP_AMD64_NEGATION: {
+			gboolean is_float = FALSE; 
+			if ( ins->inst_c1 == MONO_TYPE_R4 || ins->inst_c1 == MONO_TYPE_R8 ) 
```

Comment:
This could be just:
gboolean is_float = (ins->inst_c1 == ...

Reply:
Maybe I could move it completely to the `if` on line 11286 and not use `is_float` variable at all?